test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,594 @@
+       
+        double itrData  = g->GetValueAtIndex( (*itr).data() );
+        double gridData = g->AccessIJK( i, j, k ); 
+        if ( isNotEqual( itrData, gridData ) )
+            disagreements++;
+
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+size_t TestIterator( 
+    Grid *g, 
+    size_t& count, 
+    size_t& disagreements,
+    double& time 
+) {
+    Grid::Iterator itr;
+    Grid::Iterator enditr = g->end();
+
+    double t0 = Wasp::GetTime();
+    itr = g->begin();
+    
+    std::vector<size_t> dims = g->GetDimensions();
+
+    for ( ; itr!=enditr; ++itr) {
+        size_t i = count % dims[X];
+        size_t j = ( count / dims[X] ) % dims[Y];
+        //size_t j = count / dims[X];
+        size_t k = count / ( dims[X] * dims[Y] );
+        
+        if ( isNotEqual( *itr, g->AccessIJK( i, j, k) ) )
+            disagreements++;
+
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+size_t TestConstCoordItr( 
+    const Grid *g,
+    size_t& count, 
+    size_t& disagreements,
+    double& time 
+) {
+    Grid::ConstCoordItr itr;
+    Grid::ConstCoordItr enditr = g->ConstCoordEnd();
+
+    double t0 = Wasp::GetTime();
+    itr = g->ConstCoordBegin();
+    
+    std::vector< size_t > dims = g->GetDimensions();
+
+    for ( ; itr!=enditr; ++itr) {
+        size_t i = count % dims[X];
+        size_t j = ( count / dims[X] ) % dims[Y];
+        size_t k = count / ( dims[X] * dims[Y] );
+        size_t ijk[] = { i, j, k };
+        double coords[3];
+
+        bool disagree = false;
+        g->GetUserCoordinates( ijk, coords );
+        for ( size_t dim=0; dim<dims.size(); dim++ ) {
+            if ( isNotEqual( (*itr)[dim], coords[dim] ) ) {
+                disagree = true;
+            }
+        }
+        if ( disagree ) {
+            disagreements++;
+        }
+
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+void PrintStats( double rms, size_t numMissingValues, size_t disagreements ) {
+    cout << "    RMS error:                                     " << rms << endl;
+    cout << "    Missing value count:                           " << numMissingValues << endl;
+    cout << "    AccessIJK() vs GetValue() disagreement count:  " << disagreements << endl;
+    cout << endl;
+}
+
+bool RunTest(
+    Grid* grid
+) {
+    bool rc = 0;
+    double rms;
+    size_t numMissingValues;
+    size_t disagreements;
+
+    if ( CompareIndexToCoords( grid, rms, numMissingValues, disagreements ) ) {
+        cout << "       *** Error reported in " << grid->GetType() << " grid***" << endl;
+        rc = 1;
```

Comment:
Let's return true on success, not false :-)

Reply:
Oops.  Fixed.